jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -947,13 +947,47 @@ protected static String getResourceBaseName(Resource resource)
         StringTokenizer tokenizer = new StringTokenizer(context.getExtraClasspath(), ",;");
         while (tokenizer.hasMoreTokens())
         {
-            Resource resource = context.newResource(tokenizer.nextToken().trim());
-            String fnlc = resource.getName().toLowerCase(Locale.ENGLISH);
-            int dot = fnlc.lastIndexOf('.');
-            String extension = (dot < 0 ? null : fnlc.substring(dot));
-            if (extension != null && (extension.equals(".jar") || extension.equals(".zip")))
+            String token = tokenizer.nextToken().trim();
```

Comment:
I think the problem is that we are parsing this extraClassPath String in 2 places: once in WebAppLoader, but then again in WebInfConfiguration. It should only be parsed once. So I think the better solution - as it is WebInfConfiguration's job to set up the WebAppLoader - that it converts the extraClassPath into Resources, and then sets those resources onto WebAppLoader  in the configure() method, where it pushes the WEB-INF/lib jars and WEB-INF/classes resources into the WebAppLoader.

Reply:
There's actually 3 places.

Reply:
Will be cleaned up in 10